src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/MediationSuccessfulCallbackHandler.java

```diff
@@ -41,12 +49,47 @@ public List<CaseEvent> handledEvents() {
     }
 
     private CallbackResponse submitSuccessfulMediation(CallbackParams callbackParams) {
+        List<Element<ManageDocument>> updatedManageDocumentsList =
+            updateManageDocumentsListWithMediationAgreementDocument(callbackParams);
+
         CaseData caseDataUpdated = callbackParams.getCaseData().toBuilder()
             .businessProcess(BusinessProcess.ready(MEDIATION_SUCCESSFUL))
+            .manageDocuments(updatedManageDocumentsList)
             .build();
+
         return AboutToStartOrSubmitCallbackResponse.builder()
             .data(caseDataUpdated.toMap(objectMapper))
             .state(CASE_STAYED.name())
             .build();
     }
+
+    private List<Element<ManageDocument>> updateManageDocumentsListWithMediationAgreementDocument(
+        CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        List<Element<ManageDocument>> manageDocumentsList =
+            Optional.ofNullable(caseData.getManageDocuments()).orElse(new ArrayList<>());
```

Comment:
we are repeating code here and in the further up. perhaps this knowledge should be inside casedata. then we will just have to call that method on case data here and elsewhere where we would require it. In addition we could unit test this method on case data test